File: Sources/CountriesFlagsObjC/source/Countries/UIImage+TrinidadAndTobago.m
//
//  UIImage+TrinidadAndTobago.m
//  CountriesFlags
//

#import "Countries/UIImage+TrinidadAndTobago.h"

@implementation UIImage (TrinidadAndTobago)

+ (UIImage * _Nullable)flagTrinidadAndTobagoWithSize:(CGSize)size {
    UIColor* colorRed = [UIColor colorWithRed:206.0f/255.0f
                                        green:17.0f/255.0f
                                         blue:36.0f/255.0f
                                        alpha:1.0f];
    UIColor* colorWhite = [UIColor whiteColor];
    UIColor* colorBlack = [UIColor blackColor];
    
    UIGraphicsBeginImageContextWithOptions(size, NO, 0.0f);
    
    [colorRed setFill];
    UIRectFill(CGRectMake(0.0f, 0.0f, size.width, size.height));
    
    [colorWhite setFill];
    UIBezierPath* path = [UIBezierPath bezierPath];
    [path moveToPoint:CGPointZero];
    [path addLineToPoint:CGPointMake(size.width / 30.0f * 6.0f, 0.0f)];
    [path addLineToPoint:CGPointMake(size.width, size.height)];
    [path addLineToPoint:CGPointMake(size.width / 30.0f * 24.0f, size.height)];
    [path closePath];
    [path fill];
    
    [colorBlack setFill];
    path = [UIBezierPath bezierPath];
    [path moveToPoint:CGPointMake(size.width / 30.0f, 0.0f)];
    [path addLineToPoint:CGPointMake(size.width / 30.0f * 5.0f, 0.0f)];
    [path addLineToPoint:CGPointMake(size.width / 30.0f * 29.0f, size.height)];
    [path addLineToPoint:CGPointMake(size.width / 30.0f * 25.0f, size.height)];
    [path closePath];
    [path fill];
    
    UIImage* image = UIGraphicsGetImageFromCurrentImageContext();
    UIGraphicsEndImageContext();
    
    return image;
}

@end

File: Sources/CountriesFlags/Countries/UIImage+TrinidadAndTobago.swift
//
//  UIImage+TrinidadAndTobago.swift
//  CountriesFlags
//

#if canImport(UIKit)

import UIKit

extension UIImage {
    /**
     [wikipedia]: https://en.wikipedia.org/wiki/Flag_of_Trinidad_and_Tobago "wikipedia"
     
     Generates the flag of Trinidad and Tobago.
     For more information see [wikipedia].
     
     - Parameters:
        - size: image size.
     
     - Returns: Returns the flag of Trinidad and Tobago.
     */
    internal static func flagTrinidadAndTobago(with size: CGSize) -> UIImage? {
        let colorRed = UIColor(red: 206.0 / 255.0,
                               green: 17.0 / 255.0,
                               blue: 36.0 / 255.0,
                               alpha: 1.0)
        let colorWhite = UIColor.white
        let colorBlack = UIColor.black

        UIGraphicsBeginImageContextWithOptions(size, false, 0.0)

        colorRed.setFill()
        UIRectFill(CGRect(origin: .zero, size: size))

        colorWhite.setFill()
        let path = UIBezierPath()
        path.move(to: .zero)
        path.addLine(to: CGPoint(x: size.width / 30.0 * 6.0, y: 0.0))
        path.addLine(to: CGPoint(x: size.width, y: size.height))
        path.addLine(to: CGPoint(x: size.width / 30.0 * 24.0, y: size.height))
        path.close()
        path.fill()

        colorBlack.setFill()
        path.removeAllPoints()
        path.move(to: CGPoint(x: size.width / 30.0 * 1.0, y: 0.0))
        path.addLine(to: CGPoint(x: size.width / 30.0 * 5.0, y: 0.0))
        path.addLine(to: CGPoint(x: size.width / 30.0 * 29.0, y: size.height))
        path.addLine(to: CGPoint(x: size.width / 30.0 * 25.0, y: size.height))
        path.close()
        path.fill()

        let image = UIGraphicsGetImageFromCurrentImageContext()
        UIGraphicsEndImageContext()

        return image
    }
}

#endif
